api/docs/http_catalog.md

```diff
@@ -39,13 +39,17 @@ doubly-escaped (it should just be singly-escaped).  To keep backward compatibili
 escaping is still used/supported.
 - `transaction_uuid`: a transaction uuid identifying the entire transaction (shows up in the report as well)
 
+An optional parameter can be provided to the POST or GET to notify a node classifier that the client requested a specific
+environment, which might differ from what the client believes is its current environment.
+- `configured_environment`: the environment configured on the client
+
 ### Example Response
 
 #### Catalog found
 
-    POST /puppet/v3/catalog/elmo.mydomain.com?environment=env
+    POST /puppet/v3/catalog/elmo.mydomain.com
 
-    facts_format=pson&facts=%7B%22name%22%3A%22elmo.mydomain.com%22%2C%22values%22%3A%7B%22architecture%22%3A%22x86_64%22%7D&transaction_uuid=aff261a2-1a34-4647-8c20-ff662ec11c4c
+    environment=env&configured_environment=env&facts_format=pson&facts=%7B%22name%22%3A%22elmo.mydomain.com%22%2C%22values%22%3A%7B%22architecture%22%3A%22x86_64%22%7D&transaction_uuid=aff261a2-1a34-4647-8c20-ff662ec11c4c
```

Comment:
They can be the same. configured_environment is only nil when Puppet uses its application default (i.e. the environment wasn't configured on the client).